### PR TITLE
hlint 2.0.12 (new formula)

### PR DESCRIPTION
**Pull request by noumar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/hlint.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require "language/haskell"
+
+class Hlint < Formula
+  include Language::Haskell::Cabal
+
+  desc "Haskell source code suggestions"
+  homepage "https://github.com/ndmitchell/hlint"
+  url "https://github.com/ndmitchell/hlint/archive/v2.0.12.tar.gz"
```
Comment (ilovezfs) [inline Formula/hlint.rb]: https://hackage.haskell.org/package/hlint-2.0.12/hlint-2.0.12.tar.gz

File: Formula/hlint.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  sha256 "25cff6802f4b9c199bd4307594baa6f5647c9b3d9780c6c933fef264354bf2da"
+  head "https://github.com/ndmitchell/hlint.git"
+
+  depends_on "ghc" => :build
```
Comment (ilovezfs) [inline Formula/hlint.rb]: alphabetize

File: Formula/hlint.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on "cabal-install" => :build
+
+  def install
+    install_cabal_package :using => "happy"
+    man1.install "data/hlint.1"
+  end
+
+  test do
+    hs = testpath/"test.hs"
+    hs.write <<~EOS
```
Comment (ilovezfs) [inline Formula/hlint.rb]: (testpath/"test.hs").write <<~EOS

File: Formula/hlint.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      main = do putStrLn "Hello World"
+    EOS
+    assert_match "Warning: Redundant do", shell_output("#{bin}/hlint #{hs}", 1)
```
Comment (ilovezfs) [inline Formula/hlint.rb]: assert_match "Redundant do", shell_output("#{bin}/hlint test.hs", 1)

**Review by ilovezfs — APPROVED** [review]
some minor stuff but :+1:

**Comment by noumar** [comment]
Changes made.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @noumar! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**